src/formula.jl

```diff
@@ -109,7 +109,7 @@ function distribute!(ex::Expr)
             typeof(e)==Expr && e.head == :call && e.args[1] == distributing_op
 
         ## find first distributing subex
-        first_distributing_subex = findfirst(is_distributing_subex, ex.args)
+        first_distributing_subex = findfirst(equalto(ex.args), is_distributing_subex)
```

Comment:
`is_distributing_subex` is a predicate function so I think the original is correct.

Reply:
It might be the case, I am running the tests locally now to verify.